File: SimpleEngine/PushedMatrix.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Tao.OpenGl;

namespace SimpleEngine
{
    class PushedMatrix :  IDisposable
    {
        public PushedMatrix()
        {
            Gl.glPushMatrix();
        }

        public void Dispose()
        {
            Gl.glPopMatrix();
        }
    }
}

File: src/engine/pushedmatrix.cc
#include "engine/pushedmatrix.h"

#include "engine/opengl.h"

namespace simple
{
    PushedMatrix::PushedMatrix()
    {
        glPushMatrix();
    }

    PushedMatrix::~PushedMatrix()
    {
        glPopMatrix();
    }
}
